client/my-sites/patterns/pages/category/index.tsx

```diff
@@ -56,9 +57,22 @@ export const PatternsCategoryPage = ( {
 	const categoryNavList = categories?.map( ( category ) => ( {
 		name: category.name || '',
 		label: category.label,
-		link: `/patterns/${ category.name }`,
+		link: `/patterns/${ category.name }${ location.search }`,
```

Comment:
```suggestion
		link: isGridView ? `/patterns/${ category.name }?grid=1` : `/patterns/${ category.name }`,
```
The `useEffect` callback just above this function will strip away the search query from the URL when users navigate between categories. Even if this logic preserves the full URL query in the link, the `useEffect` callback will still override that.

Even so, I feel it's somewhat unnecessary to preserve that search query in the URL when we'll ultimately remove it anyway. 

Doing something like my suggestion would preserve just the `grid` query param instead.